src/auto/unix_socket_address.rs

```diff
@@ -0,0 +1,151 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use SocketAddress;
+use SocketConnectable;
+use UnixSocketAddressType;
+use ffi;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct UnixSocketAddress(Object<ffi::GUnixSocketAddress, ffi::GUnixSocketAddressClass>): SocketAddress, SocketConnectable;
+
+    match fn {
+        get_type => || ffi::g_unix_socket_address_get_type(),
+    }
+}
+
+impl UnixSocketAddress {
+    pub fn new(path: &str) -> UnixSocketAddress {
+        unsafe {
+            SocketAddress::from_glib_full(ffi::g_unix_socket_address_new(path.to_glib_none().0)).downcast_unchecked()
+        }
+    }
+
+    //pub fn new_abstract(path: /*Unimplemented*/&CArray TypeId { ns_id: 0, id: 10 }) -> UnixSocketAddress {
```

Comment:
Any idea why this one is not created? It's deprecated since 2.26 from what I can see, but that would always take a `&str` AFAIU

Reply:
Probably because it has a `path_len` parameter.